benchexec/tools/gazer-theta.py

```diff
@@ -0,0 +1,50 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.result as result
+import benchexec.tools.template
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for gazer-theta
+    (combined tool of Gazer and Theta)
+    https://github.com/ftsrg/gazer
+    https://github.com/ftsrg/theta
+    """
+
+    REQUIRED_PATHS = ["gazer", "gazer/tools/gazer-bmc", "gazer/tools/gazer-theta"]
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("gazer_starter.py", subdir="gazer/scripts")
+
+    def name(self):
+        return "gazer-theta"
+
+    def version(self, executable):
+        return self._version_from_tool(executable)
+
+    def cmdline(self, executable, options, task, rlimits):
+        assert len(list(task.input_files_or_identifier)) == 1
```

Comment:
Does the tool support tasks that are defined without an input file? If yes, it is fine as is, otherwise I suggest to use `task.input_files` to give a nice error message.

Reply:
Thanks, I checked out `Task` in the template and I think I understand the fields now.
Just so I get this right:
I still need the `assert len(list(task.input_files)) == 1`, because we don't support more than one input file, but I should use `task.input_files` in both the assert and the return statement, as in the case of no input files we would get and identifier and a length of 1 instead of a length with 0 in the assert if it stays like this.

Reply:
Exactly. I will add a utility method that will let you avoid the `assert`, but it is easier to add the tool-info module as you suggest and refactor it later.